### ISSUE TITLE
Merge neighbouring free zones

### ISSUE THREAD

**Issue by jgraley** [jgraley/inferno-cpp2v#721]
Following from #718, we wish to retain tree zones and just "swap in" free zones. However, _for a general free zone, it is undefined how to swap it into the tree, because it's unclear what in the tree it actually replaces_. For example there could be 3 levels of depth across multiple free zones somewhere, that replaces only 1 or 2 levels of depth in the original tree. Or 4 or more. 

We could probably match up some free zones during overlaying, but in replace-only paths, any amount of stuff could go in there (and there may be scope in overlaying to generate tree zones rather than free zones in such cases)

Rather than try to retain information from the agents which will be hard to maintain, simply join up all neighbouring free zones, repeating until every free zone only touches tree zones or root (touch is any terminus or base). Then, a tree zone can be generated indicating the zone and nodes that will be deleted when the free zone is "swapped in".

**Comment by jgraley** [comment]
Terminus numbering will be interesting, but since terminii are in DF order (after #719) the updates should be numerically simple eg a1 a2 a3 a4 into b1 b2 b3 b4 b5 b6 joining at b4 gets us b1->c1...b3->c3 then a1->c4...a4->c7 and c8->b5...c9->b6. Note that b4 is lost during the join.

**Comment by jgraley** [comment]
Need all the free zones for this, which includes duplicates of tree zones where duplication could not be avoided. So we have to have done eg #722 etc and finalised which tree zones will be duplicated.

This also indicates that we'll have to do this at execution time, which means fixing up the existing join to deal with terminii properly. Not sure we ever need to create the notional tree zone for deletion...
